Jenkinsfile

```diff
@@ -75,79 +166,77 @@ pipeline {
-                    }
-
-                    post {
-                        always{
-                            sh 'python3 resources/ci_cd/jenkins_output.py'
+                stage('Build all modes') {
+                    matrix {
+                        axes {
+                            axis {
+                                name 'buildMode'
+                                values 'release', 'debug'
+                            }
+                            axis {
+                                name 'linkMode'
+                                values 'static', 'dynamic'
+                            }
                         }
-                        success {
-                            publishChecks detailsURL: DETAILS_URL, name: STAGE_NAME,
-                                summary: ':white_check_mark: All the examples were built succesfully.',
-                                title: 'Passed', text: readFile("jenkins_output.md")
-                        }
-                        failure {
-                            publishChecks conclusion: 'FAILURE', detailsURL: DETAILS_URL,
-                                name: STAGE_NAME, title: 'Failed', text: readFile("jenkins_output.md"),
-                                summary: ':warning: There was an error building the examples.'
-                        }
-                        aborted {
-                            publishChecks conclusion: 'CANCELED', detailsURL: DETAILS_URL,
-                                name: STAGE_NAME, title: 'Aborted', text: readFile("jenkins_output.md"),
-                                summary: ':no_entry: The examples build was aborted'
+                        stages {
+                            stage('Build single mode') {
+                                steps {
+                                    echo("Build ${buildMode}/${linkMode}")
+                                    runBuildStage(buildMode, linkMode)
+                                }
+                                post {
+                                    aborted {
+                                        publishAbortedCheck(
+                                            title: 'Aborted',
+                                            summary: ':no_entry: The build was aborted.',
+                                        )
+                                    }
+                                }
+                            }
                         }
                     }
                 }
 
                 stage('Static Analysis') {
                     steps {
-                        publishChecks detailsURL: DETAILS_URL, name: STAGE_NAME,
-                            status: 'IN_PROGRESS', title: 'In progress', text: detailsText,
-                            summary: ':mag: Analysing all the examples...'
+                        publishPassedCheck(
+                            title: 'In progress',
+                            summary: ':mag: Analysing all the examples...',
```

Comment:
Analyzing

(we should use American English spelling) ... but I'm not going to hold up the pull request for this

are these customer-facing messages?

Reply:
This still hasn't changed to "Analyzing" (with a z), but I'm not going to hold up the pull request for this